components/brave_rewards/resources/creator_detection/github.ts

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import { CreatorInfo, initializeDetector } from './creator_detection'
+import { log, getPathComponents } from './helpers'
+
+initializeDetector(() => {
+
+  const nonUserPaths = new Set([
```

Comment:
FYI there are a lot more now: https://github.com/Mottie/github-reserved-names/blob/master/reserved-names.json. not sure how we can keep these up to date

Reply:
Good point.

This list is coming from the existing Greaselion scripts. It appears that this list does not need to be exhaustive - if we don't detect a "non-user" path, then the profile data fetch will presumably fail.

Reply:
again per my comment above including these in the build seems problematic. I think we still need to deliver the scripts themselves through a component